From 2cbc565fa8e79c6735f2d46d1ce3841f89579a58 Mon Sep 17 00:00:00 2001
From: ramanujadasu <vsrr.ramanujadasu@rakuten.com>
Date: Mon, 9 May 2022 14:26:52 +0530
Subject: Fixtodo testoutbound (#38797)

* fix: enable todo

* fix: update expected struct

* fix: update expected
---
 .../telemetry/outboundtrafficpolicy/helper.go  | 10 ++++++----
 .../traffic_allow_any_test.go                  | 18 ++++++++----------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/tests/integration/telemetry/outboundtrafficpolicy/helper.go b/tests/integration/telemetry/outboundtrafficpolicy/helper.go
index a3fbd4284a..724df8c959 100644
--- a/tests/integration/telemetry/outboundtrafficpolicy/helper.go
+++ b/tests/integration/telemetry/outboundtrafficpolicy/helper.go
@@ -153,10 +153,12 @@ type TestCase struct {
 // prometheus to validate that expected telemetry information was gathered;
 // as well as the http response code
 type Expected struct {
-	Query          prometheus.Query
-	StatusCode     int
-	Protocol       string
-	RequestHeaders map[string]string
+	Query           prometheus.Query
+	StatusCode      int
+	Metric          string
+	PromQueryFormat string
+	Protocol        string
+	RequestHeaders  map[string]string
 }
 
 // TrafficPolicy is the mode of the outbound traffic policy to use
diff --git a/tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go b/tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go
index 51ccf5c3ee..68bd62bc9e 100644
--- a/tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go
+++ b/tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go
@@ -178,22 +178,20 @@ func TestOutboundTrafficPolicy_AllowAny(t *testing.T) {
 			Name:     "TCP",
 			PortName: "tcp",
 			Expected: Expected{
-				// TODO(https://github.com/istio/istio/issues/22717) re-enable TCP
-				// Metric:          "istio_tcp_connections_closed_total",
-				// PromQueryFormat: `sum(istio_tcp_connections_closed_total{reporter="source",destination_service_name="PassthroughCluster",source_workload="client-v1"})`,
-				StatusCode: http.StatusOK,
-				Protocol:   "TCP",
+				Metric:          "istio_tcp_connections_closed_total",
+				PromQueryFormat: `sum(istio_tcp_connections_closed_total{reporter="source",destination_service_name="PassthroughCluster",source_workload="client-v1"})`,
+				StatusCode:      http.StatusOK,
+				Protocol:        "TCP",
 			},
 		},
 		{
 			Name:     "TCP Conflict",
 			PortName: "tcp",
 			Expected: Expected{
-				// TODO(https://github.com/istio/istio/issues/22717) re-enable TCP
-				// Metric:          "istio_tcp_connections_closed_total",
-				// PromQueryFormat: `sum(istio_tcp_connections_closed_total{reporter="source",destination_service_name="PassthroughCluster",source_workload="client-v1"})`,
-				StatusCode: http.StatusOK,
-				Protocol:   "TCP",
+				Metric:          "istio_tcp_connections_closed_total",
+				PromQueryFormat: `sum(istio_tcp_connections_closed_total{reporter="source",destination_service_name="PassthroughCluster",source_workload="client-v1"})`,
+				StatusCode:      http.StatusOK,
+				Protocol:        "TCP",
 			},
 		},
 	}
-- 
2.35.3

